source/components/triggering-changes-with-actions.md

```diff
@@ -0,0 +1,277 @@
+So far, you've learned how parent components can pass attributes in to a
+child component, and how that component can use those attributes from
+both JavaScript and its Handlebars template.
+
+But what about the opposite direction? How does data flow back out of
+the component to the parent?
+
+In Ember, components use something called **actions** to communicate
+events and changes. Actions allow you to write a function that you pass
+into a component. At the appropriate time, the component will trigger
+the action, and any behavior you put inside will be run. That code can
+do anything, from communicating with services to re-rendering the child
+component with new data.
+
+To the outside world, a component is entirely self-contained: a black
+box where attributes go in, and actions come out.
+
+### A Simple Action
+
+Let's look at a simple example of how a component can use an action to
+communicate with its parent.
+
+Imagine we work on an application where users have accounts, and we've
+been tasked with building the UI for users to delete their account.
+Because this is something we don't want users to do accidentally, we'll
+build a button that requires the user to double click it in order to
+trigger its behavior.
+
+Best of all, because components are "black boxes" that aren't coupled to
+their parent component, once we create this "double click button"
+component, we can reuse it all over our application.
+
+#### Creating the Component
+
+Let's call our component `double-click-button`. We can create it by
+typing:
+
+```text
+ember generate component double-click-button
+```
+
+Remember that you can use this component from a template like this:
+
+```hbs
+{{double-click-button}}
+```
+
+#### Designing the Action
+
+When implementing an action, it can help to break the process down into
+two steps:
+
+1. In the outside world, what do you want to happen when the action is
+   triggered?
+2. In the component, when do you trigger the action?
+
+In this example, that means:
+
+1. In the parent component, we should write an action that deletes the
+   user's account.
+2. In the button component, we should trigger the provided action when
+   the user has double clicked.
+
+Let's take it step by step.
+
+#### Implementing the Action
+
+First, we'll define what we want to happen when the user double clicks
+the button. In this case, we'll find the user's account and delete it.
+
+Where do you put the code for an action? In Ember, each component can
+have a property called `actions`, where you put functions that can be
+triggered by child components.
+
+Let's look at the parent component's JavaScript file. In this example,
+imagine we have a parent component called `user-profile` that shows the
+user's profile to them.
+
+We'll implement an action on the parent component called
+`userDidDeleteAccount()` that, when called, gets a hypothetical `login`
+service and calls the service's `deleteUser()` method.
```

Comment:
Would you be satisfied with a link to the Services section? I'm not sure how to avoid the circular dependency problem entirely. In this case, the example is sufficiently simple that I suspect must people will be able to suspend disbelief and guess at what's happening.